src/praisonai-agents/praisonaiagents/agent/tool_execution.py

```
@@ -711,6 +711,16 @@ def _trigger_after_agent_hook(self, prompt, response, start_time, tools_used=Non
             # Log learning nudge failures for debugging
             logger.warning("Learning nudge generation failed: %s", e, exc_info=True)
 
+        # Autonomous skill self-improvement loop (opt-in via self_improve=True).
+        # Runs a guarded review pass restricted to skill_manage. No-op when
+        # disabled or already inside a review (re-entrancy guarded).
+        try:
+            run_review = getattr(self, "_run_skill_review", None)
+            if run_review is not None:
+                run_review(prompt, response, tools_used)
+        except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)